shap/maskers/_output_composite.py

```diff
@@ -0,0 +1,41 @@
+from ._masker import Masker
+
+class OutputComposite(Masker):
+    def __init__(self, masker, model):
+        """ Creates a Composite masker from an underlying masker and returns the masked input along with the model output for passed args.
+
+        Parameters
+        ----------
+        masker: object
+            An object of the shap.maskers.Masker base class (eg. Text/Image masker).
+
+        model: object
+            An object shap.models.Model base class used to generate output.
+
+        Returns
+        -------
+        tuple
+            A tuple consisting of the masked input using the underlying masker appended with the model output for passed args.
+        """
+        self.masker = masker
+        self.model = model
+        # define attributes to be dynamically set
+        masker_attributes = ["shape", "invariants", "clustering", "data_transform", "mask_shapes", "feature_names"]
+        # set attributes dynamically
+        for masker_attribute in masker_attributes:
+            if getattr(self.masker, masker_attribute, None) is not None:
+                setattr(self, masker_attribute, getattr(self.masker, masker_attribute))
+
+    def __call__(self, mask, *args):
+        """ Computes mask on the args using the masker data attribute and returns tuple containing masked input with model output for passed args.
+        """
+        masked_X = self.masker(mask, *args)
+        y = self.model(*args)
+        # wrap model output
+        if not isinstance(y, tuple):
+            y = (y,)
+        # wrap masked input
+        if not isinstance(masked_X, tuple):
+            masked_X = (masked_X,)
+        return masked_X + y
+    
```

Comment:
We'd need to save() and load() functions to this, it will be pretty similar to FixedComposite masker:

`    def save(self, out_file, *args):
        super(OutputComposite, self).save(out_file)
        pickle.dump(type(self.masker), out_file)
        self.masker.save(out_file)
        self.model.save(out_file)

    @classmethod
    def load(cls, in_file):
        masker_type = pickle.load(in_file)
        if not masker_type == cls:
            print("Warning: Saved masker type not same as the one that's attempting to be loaded. Saved masker type: ", masker_type)
        return OutputComposite._load(in_file)

    @classmethod
    def _load(cls, in_file):
        masker_type = pickle.load(in_file)
        masker = masker_type.load(in_file)
        model= masker_type.load(in_file)
        outputcomposite_masker = OutputComposite(masker, model)
        return outputcomposite_masker `

Reply:
Thanks @vivekchettiar! Added the load and save functions but seems like "model= masker_type.load(in_file)" runs into an error as it encodes the model instead of the Tokenizer. 
@slundberg there are no outstanding issues with the current PR but noticed that test_gradient.py::test_pytorch_mnist_cnn test is failing (which is unrelated to changes in the PR) and when we run explainer on multiple sentences, the text plot visualization runs into error (arising because of the list of lists having varying sized output sentences).

Reply:
Thanks @ryserrao and @vivekchettiar ! I'll look into the plot error

Reply:
It turned out there were issues with the saving and loading. Specifically models needed to be supported. So I added that and now all the tests pass on my local box. Once they pass on travis then I will merge this :)